Enterprise-CMCS/macpro-mdct-mcr
# [MDCT-2628] Conditionally create new form templates

### Description
Refactor the `create` route to get or create a form template (and form template version) when the route receives a request.

Additionally, adds some types copied over from the frontend, and updates the `update` and `fetch` routes to pull form templates from a new location.

### Related ticket(s)


MDCT-2628

---
### How to test

This PR can't really be tested in the frontend until #11350 lands, but you can still sort of test it:

1. Open DynamoDB admin
2. Launch the MCR development server
3. Create a new report
4. Verify in DynamoDB admin that the new report has a version number matching the new form template version in `local-form-templates`.
5. Repeat this process and verify that no new form templates were created
6. Edit the form json of the form you created (now located in `services/app-api/forms`)
7. Save the form json and make sure the dev server output shows that it compiled.
8. Create a new form of the same type and verify that a new form template and form template version were created.

### Important updates


When #11350 lands, I'll rebase this on top of `main` and will add new testing instructions.

---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
